ap_src/ap_app/utils/switch_permissions.py

```diff
@@ -0,0 +1,68 @@
+# Models
+from ..models import (UserProfile)
+from django.contrib.auth.models import User
+
+# Utils
+from teams_utils import get_users_sharing_teams
+from user_profile import get_user_id_from_username
+from user_profile import get_userprofile_id_from_user_id
+
+# this check should be activated when the user leaves a team
+def check_for_lingering_switch_perms(username): # stops users from having switch perms when they don't share any teams
```

Comment:
This doesn't have to be changed, but just a heads up on how to comment a function in python.  You can use [docstrings](https://peps.python.org/pep-0257/).

So instead of the current comment you might have something like:

    """
    Stops users from having switch perms when they don't share any teams.
    
    Args:
        username (str): The username of the user who left the team.
    """

Reply:
That sounds like a great idea. I have never messed with docstrings before so I will take a look at changing it.

Reply:
Fixed with 509d66b